### ISSUE TITLE
lowercase @key name argument with composite sort key causes schema failure

### ISSUE THREAD

**Issue by RossWilliams** [aws-amplify/amplify-cli#2239]
**Describe the bug**
Using a @key directive with a name in lowercase and a composite sort key causes the generated query to reference an input type using incorrect capitalisation. The query input assumes a lowercase key name will stay lowercase, but the generated input type will be made uppercase.
This only fails when using a composite sort key.
**To Reproduce**
use the following test in graphql-key-transformer
```javascript
test('KeyTransformer should create named input type of correct casing when the name is lowercase', () => {
  const validSchema = `
    type Test @model @key(name: "lower", fields: ["email", "sort", "compound"], queryField: "lowercaseTest") {
        id: ID!
        email: String
        sort: String
        compound: String
    }
    `;

  const transformer = new GraphQLTransform({
    transformers: [new ModelTransformer(), new KeyTransformer()],
  });

  const { schema } = transformer.transform(validSchema);

  expect(schema).not.toContain('ModelTestlowerCompositeKeyConditionInput');
});
```
**Expected behavior**
"ModelTestlowerCompositeKeyConditionInput" should instead be generated as "ModelTestLowerCompositeKeyConditionInput".

**Additional context**
Failed after update from 1.12 -> 3.0.

Also, the test library seems to only test for statements to throw. I noticed that all statements with throw because it fails a test which checks for the @model directive being present, not because of their test scenario. So these tests currently won't protect against regressions.

**Comment by RossWilliams** [comment]
@attilah The issue looks to be making the name PascalCase here: https://github.com/aws-amplify/amplify-cli/commit/51716f340e820358087d68fd9e926084c10565eb#diff-ca2c9971cf82b7ea4207497c2bc7e611R164

Making the addCompositeSortKey function transform the name to PascalCase as well should resolve the problem.

**Comment by attilah** [comment]
@RossWilliams yeah, I encountered this issue while reviewing and testing the connections v2 PR, I think this is the same issue, but leave it open for now, connections v2 will be merged soon.

**Comment by RossWilliams** [comment]
>  I encountered this issue while reviewing and testing the connections v2 PR

If you are referring to #2230 I don't see that PR changing key transformer. Or is there another PR for me to check?

> Or you mean that addCompositeSortKey also needs the toUpper treatment?

The input type name is generated with PascalCase, so when the query is generated it needs to use the same casing as it reconstructs the input type name.

So changing addCompositeSortKey to have the same upper casing pattern should solve this

```javascript
function addCompositeSortKey(
  definition: ObjectTypeDefinitionNode,
  args: KeyArguments,
  elems: InputValueDefinitionNode[]
): InputValueDefinitionNode[] {
  let sortKeyNames = args.fields.slice(1);
  const compositeSortKeyName = toCamelCase(sortKeyNames);
  const keyName = Case.pascal(args.name || 'Primary');
  const hashKey = makeInputValueDefinition(
    compositeSortKeyName,
    makeNamedType(
      ModelResourceIDs.ModelCompositeKeyConditionInputTypeName(definition.name.value, keyName)
    )
  );
  return [hashKey, ...elems];
}
```

**Comment by attilah** [comment]
@RossWilliams found the issue, it was the same that I fixed in the PR I mentioned, just for key. I created a PR, which will be merged after Connection v2 but that release will contain this fix as well.
